src/cache/precomputed_quantities.jl

```diff
@@ -409,9 +409,10 @@ function set_precomputed_quantities!(Y, p, t)
                 Fields.field_values(Fields.level(Y.c.sgsʲs.:($j).ρae_tot, 1))
             sgsʲs_ρaq_tot_int_val =
                 Fields.field_values(Fields.level(Y.c.sgsʲs.:($j).ρaq_tot, 1))
+
+            FT = eltype(Y)
```

Comment:
```suggestion
            FT = Spaces.undertype(axes(Y.c))
```

Reply:
This could be the allocation issue? I don't think `eltype` is well defined on FieldVector's. If this is poorly inferred, it could certainly trigger allocations.

Reply:
Thank you